test/e2e/framework.go

```diff
@@ -57,12 +59,23 @@ type Framework struct {
 func NewFramework(baseName string) *Framework {
 	f := &Framework{
 		BaseName:                 baseName,
-		addonResourceConstraints: make(map[string]resourceConstraint),
+		AddonResourceConstraints: make(map[string]resourceConstraint),
+		NsCreateFunc:             createTestingNS,
 	}
+	return f.InstallGinkgoHooks()
+}
 
+// InstallGinkgoHooks installs the BeforeEach and AfterEach Ginkgo blocks
+func (f *Framework) InstallGinkgoHooks() *Framework {
 	BeforeEach(f.beforeEach)
 	AfterEach(f.afterEach)
+	return f
+}
 
+// SetNsCreateFunc allows to override the default namespace creation function
+// This function must be called before InstallGinkgoHooks() to take effect.
+func (f *Framework) SetNsCreateFunc(fn func(string, *client.Client) (*api.Namespace, error)) *Framework {
```

Comment:
a setter seems a little heavyweight for what this is doing - why not just set NsCreateFunc directly?